File: tools/bench/answers/go_haiku/s1/11_collatz_max.go
package main

import "fmt"

func main() {
	maxSteps := 0
	maxN := 0
	for n := 1; n <= 10; n++ {
		steps := 0
		x := n
		for x != 1 {
			if x%2 == 0 {
				x /= 2
			} else {
				x = 3*x + 1
			}
			steps++
		}
		if steps > maxSteps {
			maxSteps = steps
			maxN = n
		}
	}
	fmt.Printf("%d %d\n", maxN, maxSteps)
}

File: tools/bench/answers/rust_haiku/s3/11_collatz_max.rs
fn main() {
    let mut max_n = 1;
    let mut max_steps = 0;

    for n in 1..=10 {
        let mut num = n;
        let mut steps = 0;

        while num != 1 {
            if num % 2 == 0 {
                num /= 2;
            } else {
                num = 3 * num + 1;
            }
            steps += 1;
        }

        if steps > max_steps {
            max_steps = steps;
            max_n = n;
        }
    }

    println!("{} {}", max_n, max_steps);
}
